src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensHelpers.cs

```diff
@@ -120,9 +120,77 @@ internal class SemanticTokensHelpers
             var classifiedSpans = await Classifier.GetClassifiedSpansAsync(document, textSpan, cancellationToken).ConfigureAwait(false);
             Contract.ThrowIfNull(classifiedSpans, "classifiedSpans is null");
 
-            // TO-DO: We should implement support for streaming once this LSP bug is fixed:
-            // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1132601
-            return ComputeTokens(text.Lines, classifiedSpans.ToArray(), tokenTypesToIndex);
+            // Multi-line tokens are not supported by VS (tracked by https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1265495).
+            // Roslyn's classifier however can return multi-line classified spans, so we must break these up into single-line spans.
+            var updatedClassifiedSpans = ConvertMultiLineToSingleLineSpans(text, classifiedSpans.ToArray());
```

Comment:
should we change this to check the `multilineTokenSupport` client capability, and only convert if not supported?  Then we don't need to make any changes when they add support and we'll be better compliant with the lsp spec.

Reply:
I agree with David here. In general, I recommend against having any assumption of specific client behaviors in any LSP server.

Reply:
I agree this would be best, but I don't think `multilineTokenSupport` is something we're able to currently check since it isn't defined in `VSClientCapabilities`/`ClientCapabilities`. So far it's only defined in the LSP spec.

Reply:
Ah, I can't find any of the semantic tokens client capabilities defined either - https://microsoft.github.io/language-server-protocol/specification#textDocument_semanticTokens

So I'm fine with this, but we definitely need the capabilities at some point (not just the definition, but make sure the rest of the semantic tokens capabilities are set properly)

Reply:
Good point! Sounds like something we should take care of with: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1249055